### ISSUE TITLE
Audio issue at landing

### ISSUE THREAD

**Issue by attila123** [colobot/colobot#612]
I think this is an old issue coming from the original version of the game, and is not really urgent to fix, but I thought it's better to record it.
Start e.g. Terranova 4. When the doors/walls of the spaceship open up (and the camera is spinning around it) sound is correct in the left channel, but in the right channel this door opening sound can only be heard for just fractions of a second.
Even before the doors start to open, just when the spaceship lands, there are some sounds distorsions, I think.

**Comment by RaptorParkowsky** [comment]
Yep, sound implementation in GOLD is still glitchy and require refactoring, probably with new, different sound samples.

**Comment by hexagonrecursion** [comment]
The game animates both the position of the source of the  sound and the listener. Every frame it moves the source of the sound to the same position as the listener:

https://github.com/colobot/colobot/blob/8a40853234a054a73753a3afe7e80ff2e209f43f/colobot-base/src/object/auto/autobase.cpp#L1051-L1055

I have recorded OpenAL commands that move the source and the listener as well as the time between each command and was able to reproduce the effect outside of the game (it sounds slightly different because I did not recreate listener rotation). [Here is the full code](https://github.com/hexagonrecursion/sound-animate-poc/blob/main/main.cpp). Here is a sample:

```c++
    alListener3f(AL_POSITION, -117.441933, 131.421051, -91.381775);
    usleep(35022);
    alSource3f(source, AL_POSITION, -117.441933, 131.421051, -91.381775);
    usleep(2012);
    alListener3f(AL_POSITION, -113.112907, 131.401047, -96.534210);
    usleep(33395);
    alListener3f(AL_POSITION, -110.379425, 131.388412, -99.787621);
    usleep(32821);
    alSource3f(source, AL_POSITION, -110.379425, 131.388412, -99.787621);
    usleep(1510);
    alListener3f(AL_POSITION, -105.640739, 131.487839, -104.648567);
    usleep(36419);
    alListener3f(AL_POSITION, -102.642563, 131.550751, -107.724098);
    usleep(34017);
    alSource3f(source, AL_POSITION, -102.642563, 131.550751, -107.724098);
    usleep(1568);
    alListener3f(AL_POSITION, -97.480980, 131.736679, -112.282318);
    usleep(35285);
    alListener3f(AL_POSITION, -94.268082, 131.852402, -115.119644);
    usleep(41365);
    alSource3f(source, AL_POSITION, -94.268082, 131.852402, -115.119644);
```

1. Move listener to a new position
2. Wait a bunch
3. Move the source to the same position as the listener 
4. Wait just a bit
5. Move listener to a new position
6. Wait a bunch
7. ...

I believe this is the cause of the bug. I'll submit a pull to fix this soon
